webUI/app.js

```diff
@@ -289,23 +309,39 @@ function fetchPack(protocol, jsonData, packName, mcVersion) {
       }
       return response.blob();
     })
-    .then((blob) => {
+    .then(async (blob) => {
       console.log("Received pack!");
-      const url = window.URL.createObjectURL(blob);
-      const a = document.createElement("a");
-      a.style.display = "none";
-      a.href = url;
-      a.download = `${packName}.mcpack`;
-      document.body.appendChild(a);
-      a.click();
-      window.URL.revokeObjectURL(url);
+      downloadbutton.innerText = "Obtained pack!";
+      downloadbutton.classList.remove("http");
+      if (downloadbutton.classList.contains("s")) {
+        downloadbutton.classList.remove("s");
+        const url = window.URL.createObjectURL(blob);
+        const a = document.createElement("a");
+        a.style.display = "none";
+        a.href = url;
+        a.download = `${packName}.mcpack`;
+        document.body.appendChild(a);
+        a.click();
+        window.URL.revokeObjectURL(url);
+        await sleep(1000);
+        downloadbutton.innerText = "Download Selected Tweaks";
+        downloadbutton.onclick = downloadSelectedTweaks;
```

Comment:
it shouldn't be in an `if` statement, no wonder i couldnt download anything on localhost